### PR TITLE
TOOL-63: fix multiselect

### PR DESCRIPTION
**Pull request by ketewan**
## Change Summary

Locked cells break multiselect behavior since they stop click event.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

"By sense" it would be much better, if activeCell was computed upon selectedRange, but since selectedRange should be updated straightaway, and cell should become active only after child clicks are handled (to maintain old behavior, e.g. to edit date field, two clicks had to be done -- one to select cell, and another one to open date dropdown). The same could be achieved with async computed, but I didn't want to install additional library.

A little more "oddness" adds the fact, that
a) we want cell become active BEFORE dblclick handler
b) for MultiSelect data cell mouseDown event is prevented inside the component, so it's necessary to handle range both on click and on mouse down

Bug video:

https://user-images.githubusercontent.com/31967149/207728904-b5b1475b-26d1-4583-9aa0-661e05b5d47d.mov

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 1]
```diff
@@ -284,7 +282,7 @@ const { selectCell, startSelectRange, endSelectRange, clearSelectedRange, copyVa
     },
     async (ctx: { row: number; col?: number; updatedColumnTitle?: string }) => {
       const rowObj = data.value[ctx.row]
-      const columnObj = ctx.col !== null && ctx.col !== undefined ? fields.value[ctx.col] : null
+      const columnObj = ctx.col != null ? fields.value[ctx.col] : null
```
Comment (Faradey27) [inline packages/nc-gui/components/smartsheet/Grid.vue:285]: am I correct that by removing strict comparison we are try to save 1 condition? is it worth it? I am worried that on FE we started to forgot how == behavies in edge cases(like with null)

wdyt?

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 2]
```diff
@@ -296,10 +294,10 @@ const { selectCell, startSelectRange, endSelectRange, clearSelectedRange, copyVa
   )
 
 function scrollToCell(row?: number | null, col?: number | null) {
-  row = row ?? selectedCell.row
-  col = col ?? selectedCell.col
+  row = row ?? activeCell.row
+  col = col ?? activeCell.col
 
-  if (row !== undefined && col !== undefined && row !== null && col !== null) {
+  if (row != null && col != null) {
```
Comment (Faradey27) [inline packages/nc-gui/components/smartsheet/Grid.vue:300]: similar note here

File: packages/nc-gui/composables/useMultiSelect/cellRange.ts [hunk 3]
```diff
@@ -4,51 +4,53 @@ export interface Cell {
...
 
-  get end() {
+  get end(): Cell {
+    if (this.isEmpty()) {
+      return { row: null, col: null }
+    }
     return {
-      row: Math.max(this._start?.row ?? NaN, this._end?.row ?? NaN),
-      col: Math.max(this._start?.col ?? NaN, this._end?.col ?? NaN),
+      row: Math.max(this._start.row!, this._end.row!),
```
Comment (Faradey27) [inline packages/nc-gui/composables/useMultiSelect/cellRange.ts:38]: I am getting an unsafe feeling when seeing "!"

File: packages/nc-gui/composables/useMultiSelect/index.ts [hunk 4]
```diff
@@ -120,141 +123,154 @@ export function useMultiSelect(
     }
 
     return (
-      col >= selectedRange.start.col &&
-      col <= selectedRange.end.col &&
-      row >= selectedRange.start.row &&
-      row <= selectedRange.end.row
+      col >= selectedRange.start.col! &&
+      col <= selectedRange.end.col! &&
```
Comment (Faradey27) [inline packages/nc-gui/composables/useMultiSelect/index.ts:127]: here the same

File: packages/nc-gui/components/smartsheet/Grid.vue (hunk 1 above)
Comment (ketewan) [reply packages/nc-gui/components/smartsheet/Grid.vue:285]: Agree, will return back

Comment (ketewan) [reply packages/nc-gui/components/smartsheet/Grid.vue:285]: Ah, actually this value can never be undefined, so I'll just leave !== null

File: packages/nc-gui/composables/useMultiSelect/cellRange.ts (hunk 3 above)
Comment (ketewan) [reply packages/nc-gui/composables/useMultiSelect/cellRange.ts:38]: The condition check is hidden under `isEmpty`, where `isEmpty` is 
```
isEmpty() {
    return this._start.col == null || this._start.row == null || this._end.col == null || this._end.row == null
  }
```
But that, unfortunately, doesn't help with types...

**Review by vijayrathore8492 — APPROVED** [review]

**Pull request merged**